lib/reducers/create-otp-reducer.js

```diff
@@ -142,11 +143,18 @@ export function getInitialState(userDefinedConfig) {
   if (currentQuery.routingType === 'ITINERARY') {
     queryModes = ensureSingleAccessMode(queryModes)
   }
-  if (config.modes && config.modes.combinations) {
+  if (config?.modes?.combinations) {
     // Inject combinations from config for batch routing.
-    currentQuery.combinations = clone(config.modes.combinations)
+    // Filter options based on defaults if modeOptions is specified in config
+    const defaultModes = config.modes.modeOptions
+      ?.filter((m) => !m.defaultUnselected)
+      .map((m) => m.mode)
```

Comment:
Extract method and place in a file under `/lib/utils/` (maybe `itinerary.js`?). In there, when converting to TypeScript, it could be a good thing to perform type checks.